src/test/groovy/com/streamr/client/utils/OrderedMsgChainSpec.groovy

```diff
@@ -231,6 +231,29 @@ class OrderedMsgChainSpec extends Specification {
         thrown(IllegalStateException)
     }
 
+    void "empties the queue if full if skipGapsOnFullQueue is true"() {
+        int received = 0
+        OrderedMsgChain util = new OrderedMsgChain("publisherId", "msgChainId", new Consumer<StreamMessage>() {
+            @Override
+            void accept(StreamMessage streamMessage) {
+                received++
+            }
+        }, null, 5000L, 5000L, true)
+
+        when:
+        util.add(createMessage(-1, null))
+        // there's a gap between the above and the below messages, so below messages are queued
+        for (int i=1; i <= OrderedMsgChain.MAX_QUEUE_SIZE; i++) {
+            util.add(createMessage(i, i-1))
+        }
+
```

Comment:
Would be nice to assert here that the queue is full, if possible. It might make sense for sake of a well-defined test. The way it is now, this test would actually pass if there was no queue at all! :)

If that's hard to add, then never mind. I don't see this as a blocker for acceptance.